### PR TITLE
[tests] Rename NodeConn and NodeConnCB

### PR DESCRIPTION
**Pull request by jnewbery**
Final step in #11518

NodeConn -> P2PConnection
NodeConnCB -> P2PInterface

This is basically just a rename. Should be an easy review.

### CODE REVIEW

**Comment by fanquake** [comment]
After merging the only mention of `NodeConn` or `NodeConnCB` I could find was in the 0.15 release notes.

Compiled and ran the testsuite.

ACK 873beca

**Comment by practicalswift** [comment]
utACK 873beca6deda119077f53921f0d19c3ebfc7cc44

File: test/functional/example_test.py [hunk 1]
```diff
@@ -18,10 +18,11 @@
 from test_framework.mininode import (
     CInv,
     NetworkThread,
-    NodeConnCB,
+    P2PInterface,
     mininode_lock,
     msg_block,
     msg_getdata,
+    NODE_NETWORK,
```
Comment (MarcoFalke) [inline test/functional/example_test.py]: style-nit: Unused import

**Review by maflcko — COMMENTED** [review]
utACK 873beca6deda119077f53921f0d19c3ebfc7cc44

**Pull request merged**